src/PostHog.AI/PostHogOpenAIHandler.cs

```diff
@@ -0,0 +1,744 @@
+        HttpContent? content,
+        CancellationToken cancellationToken,
+        Action<Exception> onException
+    )
+    {
+        string? contentString = null;
+        JsonNode? jsonNode = null;
+
+        try
+        {
+            if (content != null)
+            {
+#pragma warning disable CA2016 // Forward cancellation token (ReadAsStringAsync has an overload with CancellationToken in .NET 5+, LoadIntoBufferAsync doesn't)
+                await content.LoadIntoBufferAsync();
+                contentString = await content.ReadAsStringAsync(cancellationToken);
+#pragma warning restore CA2016
+
+                if (!string.IsNullOrWhiteSpace(contentString))
+                {
+                    try
+                    {
+                        jsonNode = JsonNode.Parse(contentString);
+                    }
+                    catch (JsonException)
+                    {
+                        // Ignore if not JSON
+                    }
+                }
+            }
+        }
+#pragma warning disable CA1031 // Do not catch general exception types
+        catch (Exception ex)
+        {
+            onException(ex);
+        }
+#pragma warning restore CA1031
+
+        return (contentString, jsonNode);
+    }
+
+    private static string DetermineEventName(HttpRequestMessage request)
+    {
+        if (
+            request.RequestUri?.AbsolutePath.Contains(
+                "/embeddings",
+                StringComparison.OrdinalIgnoreCase
+            ) == true
+        )
+        {
+            return "$ai_embedding";
+        }
+
+        return "$ai_generation";
+    }
+
+    private async Task CaptureEventAsync(
+        HttpRequestMessage request,
+        JsonNode? requestJson,
+        HttpResponseMessage? response,
+        JsonNode? responseJson,
+        double latency,
+        string eventName,
+        Exception? exception,
+        CancellationToken cancellationToken
+    )
+    {
+        try
+        {
+            var eventProperties = new Dictionary<string, object>();
+
+            // Basic Info
+            eventProperties["$ai_provider"] = "openai";
+            eventProperties["$ai_lib"] = "posthog-dotnet";
+            eventProperties["$ai_latency"] = latency;
+
+            if (request.RequestUri != null)
+            {
+                eventProperties["$ai_base_url"] = request.RequestUri.GetLeftPart(
+                    UriPartial.Authority
+                );
+                eventProperties["$ai_request_url"] = request.RequestUri.ToString();
+            }
+
+            // Request extraction
+            string? model = null;
+            if (requestJson != null)
+            {
+                model = requestJson["model"]?.ToString();
+                eventProperties["$ai_model"] = model ?? "";
+
+                // Extract input
+                if (requestJson["messages"] is JsonArray messages)
```

Comment:
`$ai_input` and `$ai_output_choices` can contain sensitive information. We should not set these if `PostHogAIContext.PrivacyMode` is set to `true`.